server/svix-server/src/v1/endpoints/endpoint/mod.rs

```diff
@@ -216,18 +216,30 @@ impl From<EndpointHeaders> for EndpointHeadersOut {
 #[serde(rename_all = "camelCase")]
 pub struct EndpointHeadersPatchIn {
     #[validate]
-    pub headers: EndpointHeaders,
+    pub headers: EndpointHeadersPatch,
 }
 
 impl ModelIn for EndpointHeadersPatchIn {
     type ActiveModel = endpoint::ActiveModel;
 
     fn update_model(self, model: &mut Self::ActiveModel) {
+        let headers: EndpointHeaders = self
+            .headers
+            .0
+            .iter()
+            .filter_map(|(k, v)| v.as_ref().map(|v| (k.clone(), v.clone())))
+            .collect();
+
         model.headers = if let Some(Some(mut hdrs)) = model.headers.take() {
-            hdrs.0.extend(self.headers.0);
+            hdrs.0.extend(headers.0);
+            for (k, v) in self.headers.0.iter() {
+                if v.is_none() {
+                    hdrs.0.remove(k);
+                }
+            }
```

Comment:
Isn't there a way to filter a hashmap directly (functionally) rather than imperatively?
Or I'm also thinking at this point that just dropping extend from above and having:
```rust
for (k, v) in haeders.0 {
    if v.is_none() { remove(k) }
    else { add(k, v) }
```

Reply:
In my understanding we need the `.extend()` because that is what's combining existing headers (which may not be in the PATCH request) with the headers that are in the PATCH request. I might be wrong, but from my reading of the Python code we wanted to combine those two _and then_ remove anything explicitly set to NULL, as to not constrain the new headers values to only be what was in the PATCH request.

I've updated the code so that the block you commented on is performed using an `iter().for_each()`, lmk.

Reply:
Ok so are you okay with this breaking from how the Python implementation behaves? The Python implementation preserves preexisting headers that are not mentioned in the PATCH request. This suggested change would overwrite those headers with only what is mentioned in the PATCH request.

Reply:
Ah yeah I see, misread it on the first go around.